multimodal/src/autogluon/multimodal/utils/metric.py

```diff
@@ -231,15 +232,18 @@ def compute_score(
         return metric.compute(references=metric_data[Y_TRUE], predictions=metric_data[Y_PRED])
 
     metric = get_metric(metric)
-    if metric.name in [ROC_AUC, AVERAGE_PRECISION]:
-        return metric._sign * metric(metric_data[Y_TRUE], metric_data[Y_PRED_PROB][:, pos_label])
-    elif metric.name in [F1]:  # only for binary classification
-        return f1_score(metric_data[Y_TRUE], metric_data[Y_PRED], pos_label=pos_label)
+    if metric.needs_proba or metric.needs_threshold:
+        y_pred = metric_data[Y_PRED_PROB][:, pos_label]
     else:
-        try:
-            return metric._sign * metric(metric_data[Y_TRUE], metric_data[Y_PRED], y_prob=metric_data[Y_PRED_PROB])
-        except:
-            return metric._sign * metric(metric_data[Y_TRUE], metric_data[Y_PRED])
+        y_pred = metric_data[Y_PRED]
+
+    if metric.name == F1:  # only for binary classification
+        y = (metric_data[Y_TRUE] == pos_label).astype(int)
+        y_pred = (y_pred == pos_label).astype(int)
+    else:
+        y = metric_data[Y_TRUE]
```

Comment:
Put this under the `else` statement above, since it only applies when not `y_pred_proba`.

Move `y = metric_data[Y_TRUE]` to the top of the function.